server/neptune/gateway/event/comment_handler.go

```diff
@@ -68,18 +80,49 @@ func (p *CommentEventWorkerProxy) Handle(ctx context.Context, request *http.Buff
 		return p.forwardToSns(ctx, request)
 	}
 
-	if shouldAllocate && command.ForceApply {
-		p.logger.InfoContext(ctx, "running force apply command")
-		if commentErr := p.vcsClient.CreateComment(event.BaseRepo, event.PullNum, warningMessage, ""); commentErr != nil {
-			p.logger.ErrorContext(ctx, commentErr.Error())
+	if shouldAllocate {
+		if cmd.ForceApply {
+			p.logger.InfoContext(ctx, "running force apply command")
+			if err := p.vcsClient.CreateComment(event.BaseRepo, event.PullNum, warningMessage, ""); err != nil {
+				p.logger.ErrorContext(ctx, err.Error())
+			}
+			return p.scheduler.Schedule(ctx, func(ctx context.Context) error {
+				return p.forceApply(ctx, event)
+			})
 		}
-		return p.scheduler.Schedule(ctx, func(ctx context.Context) error {
-			return p.forceApply(ctx, event)
-		})
+
+		// notify user that apply command is deprecated on platform mode
+		if cmd.Name == command.Apply {
+			p.handleLegacyApplyCommand(ctx, event, cmd)
+		}
+
 	}
 	return p.forwardToSns(ctx, request)
 }
 
+func (p *CommentEventWorkerProxy) handleLegacyApplyCommand(ctx context.Context, event Comment, cmd *command.Comment) {
+	p.logger.InfoContext(ctx, "running legacy apply command on platform mode")
+
+	// appending legacy to command name to distinguish between legacy and current apply templates
+	commonData := markdown.CommonData{
+		Command: command.GetLegacyCommandTitle(cmd.Name),
+	}
+	tmpl := p.templateResolver.Resolve(commonData, event.BaseRepo, 0, 0, 0, 0)
+	if tmpl == nil {
+		p.logger.ErrorContext(ctx, "no template matched", map[string]interface{}{
+			"template": commonData.Command,
+		})
+		return
+	}
```

Comment:
Since we have default templates for everything this should never happen, I'd rather panic tbh.

Reply:
So panic here, and catch and log the error in the calling function and forward the request to sns, or propagate the error upwards and don't forward the request to sns? I'm not sure what the best practice is but if we plan to forward the  request forwards, we can just log the error here so that this function does not have to return anything and the msg will be forwarded no matter what. However, if we return an error, the calling function has to check the error, log it and still continue to forward the request which just adds more layers I think and could cause confusion for the reader. What do you think?